packages/grid/src/mouse-handlers/GridLinkMouseHandler.ts

```diff
@@ -0,0 +1,247 @@
+  width: number;
+  height: number;
+  value: string;
+  href: string;
+}
+
+class GridLinkMouseHandler extends GridMouseHandler {
+  timeoutId: undefined | ReturnType<typeof setTimeout> = undefined;
+
+  isHold = false;
+
+  isDown = false;
+
+  /**
+   * Checks if mouse is currently over a link and returns its information if so
+   * @param gridPoint Mouse position information
+   * @param grid The grid
+   * @returns A LinkInformation object if it exists and null otherwise
+   */
+  static isHoveringLink(
+    gridPoint: GridPoint,
+    grid: Grid
+  ): LinkInformation | null {
+    const { column, row, x, y } = gridPoint;
+    const { props, canvasContext: context, renderer, metrics } = grid;
+    const { model } = props;
+
+    if (row == null || column == null) {
+      return null;
+    }
+
+    assertNotNull(context);
+    assertNotNull(metrics);
+    const modelRow = grid.getModelRow(row);
+    const modelColumn = grid.getModelColumn(column);
+    const renderState = grid.getRenderState();
+    const theme = grid.getTheme();
+
+    const text = model.textForCell(modelColumn, modelRow);
+    const {
+      width: textWidth,
+      x: textX,
+      y: textY,
+    } = renderer.getTextRenderMetrics(context, renderState, column, row);
+    const {
+      fontWidths,
+      gridX,
+      gridY,
+      width: gridWidth,
+      height: gridHeight,
+      verticalBarWidth,
+      horizontalBarHeight,
+    } = metrics;
+
+    // Set the font and change it back after
+    const prevFont = context.font;
+    context.font = theme.font;
+
+    const fontWidth =
+      fontWidths.get(context.font) ?? GridRenderer.DEFAULT_FONT_WIDTH;
+    const truncationChar = model.truncationCharForCell(modelColumn, modelRow);
+    const truncatedText = GridRenderer.truncateToWidth(
+      context,
+      text,
+      textWidth,
+      fontWidth,
+      truncationChar
+    );
+
+    // Check if the truncated text contains a link
+    if (linkify.find(truncatedText, 'url').length !== 0) {
+      const tokenizedText = model.tokensForCell(truncatedText);
+      const {
+        actualBoundingBoxAscent,
+        actualBoundingBoxDescent,
+      } = context.measureText(truncatedText);
+      const linkHeight = actualBoundingBoxAscent + actualBoundingBoxDescent;
+      // The x position of the word
+      let startX = textX;
+
+      // Consider edge cases
+      const top = Math.max(gridY, gridY + textY - linkHeight / 2);
+      const bottom = Math.min(
+        top + linkHeight,
+        gridHeight - horizontalBarHeight
+      );
+
+      // Loop through the blocks of text
+      for (let i = 0; i < tokenizedText.length; i += 1) {
+        const { v: value } = tokenizedText[i];
+        let { width: linkWidth } = context.measureText(value);
+
+        // If the text is not visible (right side is less than gridX)
+        if (startX + linkWidth < gridX) {
+          startX += linkWidth;
+
+          // eslint-disable-next-line no-continue
+          continue;
+        }
```

Comment:
Do you mean in `setCursor`? Sorry, could you clarify what you mean?

Reply:
On L129 there's a `if (x >= left && x <= right &&...` check. You could also include this condition there instead of needing this check and the continue.

I would recommend combining some of these into booleans before the check just for readability. It can get tricky to understand all the checks in some cases otherwise

```
const cursorIsInBounds = x >= left && x <= right && y >= top && y <= bottom;
const textIsVisible = startX + linkWidth >= gridX;

if (cursorIsInBounds && textIsVisible) {
  return { stuff };
}
```